controllers/datadogagent/override/podtemplatespec.go

```diff
@@ -6,22 +6,177 @@
 package override
 
 import (
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/util/errors"
+	"fmt"
+	"strings"
 
+	apicommon "github.com/DataDog/datadog-operator/apis/datadoghq/common"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/common/v1"
 	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
 	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object/volume"
+	corev1 "k8s.io/api/core/v1"
 )
 
 // PodTemplateSpec use to override a corev1.PodTemplateSpec with a 2alpha1.DatadogAgentPodTemplateOverride.
-func PodTemplateSpec(manager feature.PodTemplateManagers, override *v2alpha1.DatadogAgentComponentOverride) (*corev1.PodTemplateSpec, error) {
-	// TODO(operator-ga): implement OverridePodTemplate
+func PodTemplateSpec(manager feature.PodTemplateManagers, override *v2alpha1.DatadogAgentComponentOverride, componentName v2alpha1.ComponentName, ddaName string) {
+	// Notice that there are several attributes in a
+	// v2alpha1.DatadogAgentComponentOverride, like "Replicas" or "Disabled",
+	// that are not related with a pod template spec. The overrides for those
+	// attributes are not applied in this function.
+
+	if override == nil {
+		return
+	}
+
+	// TODO: seccomprootpath, seccompcustomprofile, seccomprofilename.
+	// They should only apply to system-probe, and I think there's some setup
+	// that we need to do before being able to override them.
+
+	if override.Name != nil {
+		manager.PodTemplateSpec().Name = *override.Name
+	}
+
+	if override.ServiceAccountName != nil {
+		manager.PodTemplateSpec().Spec.ServiceAccountName = *override.ServiceAccountName
+	}
+
+	if override.Image != nil {
+		for i, container := range manager.PodTemplateSpec().Spec.Containers {
+			manager.PodTemplateSpec().Spec.Containers[i].Image = overriddenImage(container.Image, override.Image)
+		}
+
+		for i, initContainer := range manager.PodTemplateSpec().Spec.InitContainers {
+			manager.PodTemplateSpec().Spec.InitContainers[i].Image = overriddenImage(initContainer.Image, override.Image)
+		}
+	}
+
+	for _, env := range override.Env {
+		manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+			Name:  env.Name,
+			Value: env.Value,
+		})
+	}
+
+	overrideCustomConfigs(manager, override.CustomConfigurations, componentName, ddaName)
+
+	// Note: override.ExtraConfd.ConfigData requires creating a configmap, so it cannot be handled here
+	if override.ExtraConfd != nil && override.ExtraConfd.ConfigMap != nil {
```

Comment:
could it be more generic like: override.AdditionalVolumes. 
and the customer add a configmap itself. and then it need to add a VolumeMount in the containerOverride?

wdyt?

Reply:
my :2cents: we have `override.Volumes` below; i think it might be beneficial to keep `override.ExtraConfd` to make this option transparent and be aligned with the helm chart